packages/core/src/controllers/SendController.ts

```diff
@@ -93,49 +94,85 @@ export const SendController = {
+                W3mFrameRpcConstants.ACCOUNT_TYPES.SMART_ACCOUNT,
+              token: this.state.token.address,
+              amount: this.state.sendTokenAmount,
+              network: NetworkController.state.caipNetwork?.id || ''
+            }
+          })
+          this.sendERC20Token({
+            receiverAddress: this.state.receiverAddress,
+            tokenAddress: this.state.token.address,
+            sendTokenAmount: this.state.sendTokenAmount,
+            decimals: this.state.token.quantity.decimals
+          })
+        } else if (
+          this.state.receiverAddress &&
+          this.state.sendTokenAmount &&
+          this.state.gasPrice &&
+          this.state.token?.quantity.decimals
+        ) {
+          EventsController.sendEvent({
+            type: 'track',
+            event: 'SEND_INITIATED',
+            properties: {
+              isSmartAccount:
+                AccountController.state.preferredAccountType ===
+                W3mFrameRpcConstants.ACCOUNT_TYPES.SMART_ACCOUNT,
+              token: this.state.token?.symbol,
+              amount: this.state.sendTokenAmount,
+              network: NetworkController.state.caipNetwork?.id || ''
+            }
+          })
+          this.sendNativeToken({
+            receiverAddress: this.state.receiverAddress,
+            sendTokenAmount: this.state.sendTokenAmount,
+            gasPrice: this.state.gasPrice,
+            decimals: this.state.token.quantity.decimals
+          })
         }
-      })
-      this.sendERC20Token({
-        receiverAddress: this.state.receiverAddress,
-        tokenAddress: this.state.token.address,
-        sendTokenAmount: this.state.sendTokenAmount,
-        decimals: this.state.token.quantity.decimals
-      })
-    } else if (
-      this.state.receiverAddress &&
-      this.state.sendTokenAmount &&
-      this.state.gasPrice &&
-      this.state.token?.quantity.decimals
-    ) {
-      EventsController.sendEvent({
-        type: 'track',
-        event: 'SEND_INITIATED',
-        properties: {
-          isSmartAccount:
-            AccountController.state.preferredAccountType ===
-            W3mFrameRpcConstants.ACCOUNT_TYPES.SMART_ACCOUNT,
-          token: this.state.token?.symbol,
-          amount: this.state.sendTokenAmount,
-          network: NetworkController.state.caipNetwork?.id || ''
+
+        return
+      case 'solana':
+        if (!this.state.sendTokenAmount || !this.state.receiverAddress) {
+          SnackController.showError('Please enter a valid amount and receiver address')
+
+          return
         }
-      })
-      this.sendNativeToken({
-        receiverAddress: this.state.receiverAddress,
-        sendTokenAmount: this.state.sendTokenAmount,
-        gasPrice: this.state.gasPrice,
-        decimals: this.state.token.quantity.decimals
-      })
+
+        RouterController.pushTransactionStack({
+          view: 'Account',
+          goBack: false
+        })
+
+        // Implement solana
```

Comment:
should we remove this commetn?

Reply:
https://github.com/WalletConnect/web3modal/pull/2802/commits/96567da7a5f4157adec07760dca3bfba8fcb3b06